### PR TITLE
[TEST](api_gtest): add psroipool api cases and modify roi_crop destroy

### PR DESCRIPTION
**Pull request by LittleReal**

### CODE REVIEW

File: bangc-ops/test/mlu_op_gtest/api_gtest/src/gtest/psroipool_forward/psroipool_forward.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+*************************************************************************/
+#include <iostream>
+#include <vector>
+#include <string>
+#include <tuple>
+#include "api_test_tools.h"
+#include "core/logging.h"
+#include "core/tensor.h"
+#include "gtest/gtest.h"
+#include "mlu_op.h"
```
Comment (tudejiang79) [inline bangc-ops/test/mlu_op_gtest/api_gtest/src/gtest/psroipool_forward/psroipool_forward.cpp:20]: #include <iostream>
#include <vector>
#include <string>
#include <tuple>

#include "api_test_tools.h"
#include "core/logging.h"
#include "core/tensor.h"
#include "gtest/gtest.h"
#include "mlu_op.h"

File: bangc-ops/test/mlu_op_gtest/api_gtest/src/gtest/psroipool_forward/psroipool_forward_general.cpp [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                                         4, std::vector<int>({1, 4, 4, 9}))),
+        testing::Values(MLUOpTensorParam(MLUOP_LAYOUT_ARRAY, MLUOP_DTYPE_FLOAT,
+                                         2, std::vector<int>({2, 5}))),
+        testing::Values(MLUOpTensorParam(MLUOP_LAYOUT_NHWC, MLUOP_DTYPE_FLOAT,
+                                         4, std::vector<int>({2, 3, 3, 1}))),
+        testing::Values(MLUOpTensorParam(MLUOP_LAYOUT_NHWC, MLUOP_DTYPE_INT32,
+                                         4, std::vector<int>({2, 3, 3, 1}))),
+        testing::Values(MLUOP_UNKNOWN_DEVICE),
+        testing::Values(MLUOP_STATUS_BAD_PARAM)));
+}  // mluopapitest
```
Comment (tudejiang79) [inline bangc-ops/test/mlu_op_gtest/api_gtest/src/gtest/psroipool_forward/psroipool_forward_general.cpp:315]: 少空格

**Review by tudejiang79 — APPROVED** [review]

**Review by ling08 — APPROVED** [review]

**Review by guzhonghao — APPROVED** [review]
LGTM

**Review by ling08 — APPROVED** [review]
LGTM

**Review by WangYuan-cn — APPROVED** [review]

**Pull request merged**